showdown/engine/damage_calculator.py

```diff
@@ -236,16 +236,19 @@ def weather_modifier(attacking_move, weather):
         return 1.5
     elif weather == constants.RAIN and attacking_move[constants.TYPE] == 'fire':
         return 0.5
+    elif weather == constants.HEAVY_RAIN and attacking_move[constants.TYPE] == 'fire':
+        return 0
+    elif weather == constants.HEAVY_RAIN and attacking_move[constants.TYPE] == 'water':
+        return 1.5
     elif weather == constants.DESOLATE_LAND and attacking_move[constants.TYPE] == 'water':
         return 0
+    elif weather == constants.DESOLATE_LAND and attacking_move[constants.TYPE] == 'fire':
+        return 1.5
     return 1
 
 
 def stab_modifier(attacking_pokemon, attacking_move):
-    if attacking_move[constants.TYPE] in [t for t in attacking_pokemon.types]:
-        return 1.5
-
-    return 1
+    return 1.5
```

Comment:
Sorry I probably wasn't clear here - there's no change that should be done here.

This should just remain as it is on the master branch. 

There are a bunch of tests failing because this change gives every move a 1.5x STAB boost,

Reply:
thanks, i've just fixed that !